### ISSUE TITLE
measurementId not set on firebase.initializeApp()

### ISSUE THREAD

**Issue by AaronBentley** [alexluong/gatsby-packages#17]
@alexluong Thank you for your efforts on this awesome plugin , loving the updates in 0.2.0-beta.3 👏

More of a heads up than an issue - whilst migrating I noticed the measurementId is not set as it's not provided on `firebase.initializeApp()` - which makes `firebase/analytics` really cranky, like, big red text cranky 😱

<img width="823" alt="Screenshot 2020-03-03 at 22 18 03" src="https://user-images.githubusercontent.com/3936200/75825300-f8e10800-5d9c-11ea-98ca-15db2a16d985.png">

**Comment by alexluong** [comment]
Thanks for reporting the issue @AaronBentley. This is definitely an oversight on my part. Published another version `0.2.0-beta.4` with a fix. Let me know if this is still an issue!

**Comment by AaronBentley** [comment]
Thank you @alexluong, confirmed as a fix ✅

I may have found another, wasn't sure whether to raise it as an additional issue or not - after installing 0.2.0-beta.4 and adding...

```javascript
import 'firebase/analytics'
import 'firebase/functions'
import 'firebase/performance'
```

...to both `gatsby-browser.js`& `Gatsby-ssr.js` - running `gatsby develop` fails with the following error: 

<img width="950" alt="Screenshot 2020-03-03 at 22 50 29" src="https://user-images.githubusercontent.com/3936200/75827376-6abb5080-5da1-11ea-9fe5-cc01a03979e1.png">

However, I've found a workaround of sorts which I hope helps better debug this:

1. Comment out/remove any `firebase/*` import statements in both `gatsby-browser.js` and `gatsby-ssr.js`.
2. Run `gatsby develop` - which builds successfully.
3. Then return to `gatsby-browser.js` and `gatsby-ssr.js`to uncomment/add back the imports removed in step 1.

Any ideas what could be throwing it on on the initial build?

**Comment by alexluong** [comment]
@AaronBentley I can't seem to reproduce this issue. Can you create a codesandbox or a sample repo with just `gatsby-plugin-firebase` with this issue?

**Comment by AaronBentley** [comment]
Thanks for investigating @alexluong, I've created [an example on CodeSandbox](https://codesandbox.io/s/gatsby-plugin-firebase-diagnostics-j9h3p?fontsize=14&hidenavigation=1&theme=dark) as requested.

In this sandbox, I'm getting a different 'IDB' error on the reproduced case...

![Screenshot 2020-03-04 at 09 51 36](https://user-images.githubusercontent.com/3936200/75867040-e34ffa80-5dfd-11ea-8611-fd9cb621ffe6.png)

However I've noticed the error only appears when using Safari (v12.1.1), Chrome seems to be working as intended - additionally, the error doesn't happen if I check 'Disable Cross-Origin Restrictions' in the Safari Develop menu. 🤔

Could this be a browser specific issue perhaps?
